project_report/project_information.py

```diff
@@ -0,0 +1,489 @@
+        return project_size / 1099511627776.0
+
+    @staticmethod
+    def parse_date(date):
+        if not date:
+            return 'NA'
+        return datetime.datetime.strptime(date, '%Y-%m-%dT%H:%M:%S.%f').strftime('%d %b %y')
+
+    @staticmethod
+    def calculate_mean(values):
+        return sum(values) / max(len(values), 1)
+
+    @property
+    def project_title(self):
+        return self.project.udf.get('Project Title', '')
+
+    @property
+    def quote_number(self):
+        return self.project.udf.get('Quote No.', '')
+
+    @property
+    def enquiry_number(self):
+        return self.project.udf.get('Enquiry Number', '')
+
+    @property
+    def number_quoted_samples(self):
+        return self.project.udf.get('Number of Quoted Samples', '')
+
+    @property
+    def customer_name(self):
+        name = self.project.researcher.lab.name
+        if not name:
+            name = self.project.researcher.name
+        return name
+
+    @property
+    def customer_address_lines(self):
+        address_keys = ('Shipment Address Line 1', 'Shipment Address Line 2', 'Shipment Address Line 3',
+                        'Shipment Address Line 4', 'Shipment Address Line 5')
+        return [
+            self.project.udf.get(k)
+            for k in address_keys
+            if self.project.udf.get(k) and self.project.udf.get(k) != '-'
+        ]
+
+    def update_from_program_csv(self, program_csv):
+        all_programs = {}
+        if program_csv and path.exists(program_csv):
+            with open(program_csv) as open_prog:
+                for row in csv.reader(open_prog):
+                    all_programs[row[0]] = row[1]
+        # TODO: change the hardcoded version of bcl2fastq
+        #all_programs['bcl2fastq'] = '2.17.1.14'
+        for p in ['bcl2fastq', 'bcbio', 'bwa', 'gatk', 'samblaster']:
+            if p in all_programs:
+                self.params[p + '_version'] = all_programs.get(p)
+
+    @staticmethod
+    def get_from_project_summary_yaml(summary_yaml):
+        with open(summary_yaml, 'r') as open_file:
+            full_yaml = yaml.safe_load(open_file)
+        sample_yaml = full_yaml['samples'][0]
+        return path.basename(path.dirname(sample_yaml['dirs']['galaxy'])), sample_yaml['genome_build']
+
+    def get_bcl2fastq_version(self, run_ids):
+        bcl2fastq_versions = set()
+        for run_id in run_ids:
+            prog_vers_yaml = os.path.join(self.run_folders, run_id, 'program_versions.yaml')
+            with open(prog_vers_yaml, 'r') as open_file:
+                full_yaml = yaml.safe_load(open_file)
+                if 'bcl2fastq' in full_yaml and full_yaml.get('bcl2fastq'):
+                    bcl2fastq_versions.add(full_yaml.get('bcl2fastq'))
+                else:
+                    self.warning('Run %s has no bcl2fastq version: default to v2.17.1.14', run_id)
+                    bcl2fastq_versions.add('v2.17.1.14')
```

Comment:
EdinburghGenomics/Analysis-Driver#369 will soon make this incorrect

Reply:
The bcl2fastq version will be extracted from the `program_versions.yaml` in the run directory so we should get the updated value. only if the yaml file is missing are we reverting back to the default version.

Reply:
All runs processed since March last year have all had the bcl2fastq version in program_vesrions.yaml, so we should always be getting the correct reported version. This should work for now but for the record, I do not like that we're doing this.

Reply:
Alternative is that we break at this point.
It means that if we redeliver old data we will have to create the yaml file manually.

Reply:
I agree with breaking at this point if the version is not defined in the yaml. It would provide whoever is running the process with the opportunity to correct the omission.

Reply:
Added exception when the file is not present or the version of bcl2fastq is not found